### PR TITLE
[Design Review] Upload

### LINKED ISSUE #456: [Task] [Styling] Address underline under link requiring being nested
https://github.com/cfpb/sbl-frontend/pull/423#issuecomment-2097105437

### LINKED ISSUE #467: [Upload] Implement an Alert and body text to address SUBMISSION_ACCEPTED
## Explanation
- The user clicked on the Warnings accepted confirmation -- status is now SUBMISSION_ACCEPTED -- but if the user goes back to upload there isn't a message to notify the user of this

### LINKED ISSUE #473: [Upload] Excel & Figma content pass

### LINKED ISSUE #478: [Point of Contact] [Select] State Selector has a double value
Screenshot 2024-05-09 at 10 25 40 AM

### LINKED ISSUE #480: [Design Review] Upload

### PR DESCRIPTION
**Pull request by shindigira**
closes #480 
closes #473 
closes #456
closes #467 
closes #478 

## Note
- `yarn install` to pull in the DSR changes to the pagination component
- Screenshots show a checkmark instead of an exclamation mark during errors/warnings. This is changed

## Changes
- enhancement(Upload): Updated content to match the excel and figma
- feat(Upload): initial load (get submission latest) 4xx/5xx's redirect to the 500 page
- style(Upload Alerts - Links): underlined
- style(Upload): Updated based on https://github.com/cfpb/sbl-frontend/pull/430#issuecomment-2103750823
- ~~enhancement(download report): removed caching headers -- using `submissionId` so the browser caching is fine~~

## Extra note 

- The two scenarios not added to this design review:
  - VALIDATION_EXPIRED (Needs a big file to test the two minute limit)
  - If initial GET LATEST SUBMISSION fails, redirect to 500 page 

- The button changes are not in yet

### Introduction
Screenshot 2024-05-10 at 3 01 16 PM

### Never uploaded before
Screenshot 2024-05-10 at 1 17 22 PM

### Pass
[pass_123456789TESTBANK456_small.csv](https://github.com/cfpb/sbl-frontend/files/15223922/pass_123456789TESTBANK456_small.csv)
Screenshot 2024-05-10 at 2 32 07 PM

### Warnings
[logic-warnings_small.csv](https://github.com/cfpb/sbl-frontend/files/15223933/logic-warnings_small.csv)
Screenshot 2024-05-10 at 2 32 53 PM

### Errors
[logic-errors_small.csv](https://github.com/cfpb/sbl-frontend/files/15223957/logic-errors_small.csv)
Screenshot 2024-05-10 at 2 33 27 PM

### SUBMISSION_ACCEPTED - no warnings
[pass_123456789TESTBANK456_small.csv](https://github.com/cfpb/sbl-frontend/files/15223922/pass_123456789TESTBANK456_small.csv)
Screenshot 2024-05-10 at 2 58 09 PM

### SUBMISSION_ACCEPTED - warnings
[logic-warnings_small.csv](https://github.com/cfpb/sbl-frontend/files/15280368/logic-warnings_small.csv)
Screenshot 2024-05-10 at 2 59 00 PM

### File size upload limit (Can use 2MB test limit -- normally 2GB)
[logic-errors_medium.csv (2MB)](https://github.com/cfpb/sbl-frontend/files/15223975/logic-errors_medium.csv)
Screenshot 2024-05-10 at 2 53 15 PM

### Incorrect file type (renamed to `.csv) -- General Catch-All Validation Error
[alf-picture copy.csv](https://github.com/cfpb/sbl-frontend/files/15224012/alf-picture.copy.csv)
Screenshot 2024-05-10 at 2 54 56 PM

### Incorrect file type (example `.png`)
[alf-picture](https://github.com/cfpb/sbl-frontend/assets/13324863/16440bf1-6358-49d4-8034-9e66cd53596a)
Screenshot 2024-05-10 at 2 59 40 PM

### Submission Malformed
[format-error__missing-more-columns.csv](https://github.com/cfpb/sbl-frontend/files/15267001/format-error__missing-more-columns.csv)
Screenshot 2024-05-10 at 3 00 07 PM

### CODE REVIEW

**Comment by billhimmelsbach** [comment]
@natalia-fitzgerald 

TODO:
- The backend/frontend errors are almost reviewed, @angelcardoz is going to take a peek at them

**Comment by natalia-fitzgerald** [comment]
@shindigira 
I took a look at the screenshots posted above and I have some initial feedback in advance of our design review.  
- Check the vertical space between the body text and the "Replace file" button. It looks like the space in your image is 30px instead of the 15px shown in the mock-up 
- A number of the inline links in the error messages shown above are missing the underlines
- Update intro copy and body text to match the content shown in Figma
- Update success and errors form and field level alerts to the content shown in the "Upload and validation messages" spreadsheet
- Vertical space between bottom of well and navigation buttons should be 30px
- This may be right already but I wanted to double check -- the space above and below the horizontal rule line in the well should be 30px
- Navigation at the bottom of the page should only include the single "Save and continue"

File: src/pages/ProfileForm/Step1Form/Step1FormHeader.tsx [hunk 1]
```diff
@@ -24,8 +24,10 @@ function Step1FormHeader({ isStep1 }: Step1FormHeaderProperties): JSX.Element {
           <>
             In order to begin using the platform you must have a Legal Entity
             Identifier (LEI) for your financial institution. Visit the{' '}
-            <Link href={gleifLink}>Global LEI Foundation (GLEIF)</Link> website
-            for more information on how to obtain an LEI.
+            <Link className='border-b-[1px] border-dotted' href={gleifLink}>
+              Global LEI Foundation (GLEIF)
+            </Link>{' '}
+            website for more information on how to obtain an LEI.
           </>
```
Comment (meissadia) [inline src/pages/ProfileForm/Step1Form/Step1FormHeader.tsx:31]: I think when we change these fragments to `<p>` elements, the link styling takes care of itself - so wouldn't need to pass classNames.

**Review by meissadia — APPROVED** [review]
This is fantastic! Great work, amazing attention to detail, and impressive job keeping all these variants straight.

**Comment by natalia-fitzgerald** [comment]
@shindigira 
This is looking great! I think we're getting close. I'll check AWS next but here's feedback based on the screenshots. 

A few content updates to the base page: 
  - Update content intro content (large and small text blocks) to match Figma. 
  - Update the content within the well to match Figma

I compared Figma to the screenshots and noticed some missing scenarios in the screenshots. I know that you mentioned missing ones in your post above but just so that I can be sure we're referencing the same ones, these are the ones I didn't see in the images (above): 

**UPLOAD_FAILED (500 error, upload fails to reach S3 bucket)**
Screenshot 2024-05-14 at 11 14 16 PM

**Upload file (VALIDATION_EXPIRED) -- you did mention above that this one is in progress**
Screenshot 2024-05-14 at 11 15 04 PM

**Generic 400s, 500s, etc (upload)**
If a user uploads a file and the server fails for whatever reason that's not included in the errors above, then this generic message is shown.
Screenshot 2024-05-14 at 11 18 04 PM

**Comment by natalia-fitzgerald** [comment]
@shindigira 

Listing some additional changes here so they aren't missed: 

- Button text: Change "Upload your file" to "Upload file"
- Button text: Change "Replace file" to "Upload new file"
- Change "Return to Filing Overview" to "Go back" and have this go back to the previous page (even if it's Provide type of financial institution)

File: src/api/requests/fetchFilingSubmissionLatest.ts [hunk 2]
```diff
@@ -70,24 +70,28 @@ async function retryRequestWithDelay(
   // eslint-disable-next-line no-param-reassign
   axiosInstance.defaults.retryCount += One;
 
-  // TODO: Remove console.logs once retry adjustments have been fully accepted
-  console.log(
-    'Validation STILL in-progress - Long Polling - RETRYING',
-    response,
-  );
+  if (import.meta.env.DEV) {
```
Comment (billhimmelsbach) [inline src/api/requests/fetchFilingSubmissionLatest.ts:73]: Wonder how the security scan stuff will handle these dev only console.logs. We'll find out. 👍

File: src/components/Button.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ type ButtonProperties = ComponentProps<typeof ButtonDSR>;
 /* Handles new secondary styling */
 
 const newSecondaryStyle =
-  'cursor-pointer border-[1px] border-solid border-pacific bg-white text-pacific hover:border-[#0050B4] hover:bg-white hover:text-[#0050B4] focus:bg-transparent disabled:cursor-not-allowed disabled:border-none';
+  'cursor-pointer border-[1px] border-solid border-pacific bg-white text-pacific hover:border-[#0050B4] hover:bg-white hover:text-[#0050B4] focus:bg-white disabled:cursor-not-allowed disabled:border-none';
```
Comment (billhimmelsbach) [inline src/components/Button.tsx:9]: Looking forward to [getting this in the DSR](https://github.com/cfpb/design-system-react/issues/341) post-mvp.

File: src/pages/Filing/FilingApp/FileDetailsValidation.tsx [hunk 4]
```diff
@@ -6,27 +6,32 @@ import { fileSubmissionValidationStatus } from './FileSubmission.data';
...
-      dataGetSubmissionLatest.state in fileSubmissionValidationStatus
-      ? // @ts-expect-error use key
-        fileSubmissionValidationStatus[dataGetSubmissionLatest.state]
-      : '';
+  const validationStatusMessage: string =
+    hasWarnings &&
+    dataGetSubmissionLatest.state === FileSubmissionState.SUBMISSION_ACCEPTED
+      ? 'Your file contains no errors and warnings have been verified'
+      : // @ts-expect-error key in
+        dataGetSubmissionLatest.state in fileSubmissionValidationStatus
```
Comment (billhimmelsbach) [inline src/pages/Filing/FilingApp/FileDetailsValidation.tsx:31]: Oh interesting... I added the exception to [this ticket](https://github.com/cfpb/sbl-frontend/issues/528)

**Review by billhimmelsbach — APPROVED** [review]
Great! Approving this for merging so that everything can land on our preview environment, but @natalia-fitzgerald will still get the final say on the design review. ✅

I've also added a ticket to cleanup some of our Typescript exceptions for post-mvp:
https://github.com/cfpb/sbl-frontend/issues/528

**Pull request merged**